pycbc/events/stat.py

```diff
@@ -87,282 +93,214 @@ def single(self, trigs):
         numpy.ndarray
             The array of single detector values
         """
-        return ranking.get_newsnr(trigs)
+        return ranking.get_sngls_ranking_from_trigs(
+            trigs,
+            self.sngl_ranking,
+            **self.sngl_ranking_kwargs
+        )
 
-    def coinc(self, s0, s1, slide, step): # pylint:disable=unused-argument
-        """Calculate the coincident detection statistic.
+    def single(self, trigs): # pylint:disable=unused-argument
+        """
+        Calculate the necessary single detector information
 
         Parameters
         ----------
-        s0: numpy.ndarray
-            Single detector ranking statistic for the first detector.
-        s1: numpy.ndarray
-            Single detector ranking statistic for the second detector.
-        slide: (unused in this statistic)
-        step: (unused in this statistic)
+        trigs: dict of numpy.ndarrays, h5py group (or similar dict-like object)
+            Dictionary-like object holding single detector trigger information.
 
         Returns
         -------
         numpy.ndarray
-            Array of coincident ranking statistic values
+            The array of single detector values
         """
-        return (s0 ** 2. + s1 ** 2.) ** 0.5
+        err_msg = "This function is a stub that should be overridden by the "
+        err_msg += "sub-classes. You shouldn't be seeing this error!"
+        raise ValueError(err_msg)
 
-    def coinc_lim_for_thresh(self, s0, thresh):
-        """Calculate the required single detector statistic to exceed
-        the threshold for each of the input triggers.
+    # FIXME: Why does this function not just call straight into single?
+    # FIXME: Should this be renamed for clarity if multiifo is being removed?
+    def single_multiifo(self, single_info):
```

Comment:
The single_multiifo() function is to generate a ranking statistic for a single-detector candidate which is equivalent to the ranking statistic of a coincident event.

This is distinct from the single() function as that is used as the intermediate step in the likelihood-based statistics. In sum-of-squares statistics, it is entirely equivalent.

I agree that a better name should be used for this. e.g. rank_stat_single (and coinc_multiifo/coinc renamed to rank_stat_coinc)

Reply:
@GarethDaviesGW The naming is fine with me. Would you be okay with having `rank_stat_single` just call straight through to the `single` method and replace this:
https://github.com/gwastro/pycbc/blob/master/bin/hdfcoinc/pycbc_multiifo_sngls_findtrigs#L144
with just the one call?

Reply:
Yeah that makes sense

Reply:
I missed this comment. 'rank_stat_coinc' seems like an unnecessarily long name given that the module and classes in it and instances in the rest of the code already have the name 'ranking' or 'stat'.  I'd suggest 'coinc_rank' and 'single_rank'. 

Also if we're renaming things, how about renaming 'ranking.py' -> 'sngl_statistic.py' ?

Reply:
I'm happy to rename these to whatever is most useful.

I would avoid `single_rank` to avoid confusion with the stuff defined in `ranking.py`. I'm not quite sure what the right words are to distinguish between "single detector ranking using only information from the trigger itself", which is being referred to as "single ranking"  and "single detector ranking using bulk information and fits". Maybe `coinc_stat` and `single_stat`??

Reply:
For similar reasons renaming ranking.py to `sngl_statistic` confuses in my head the distinction between these two things.

Reply:
what I'm suggesting would end up with the logic that 'single detector number calculated using only trigger information' is a 'sngl stat(istic)' and things analogous to the coinc ranking stat which may use global information are called 'rank(ing)' 
i.e. something calculated at single trigger label is always a 'statistic', but in general won't be a ranking.

Reply:
and yes, this is a bit backwards on wrt the current naming of stat.py and ranking.py, but i think makes more sense. (also consistent with naming instances of a ranking-stat object as 'rank_method' in scripts)